solr/solrj/src/java/org/apache/solr/common/util/Utils.java

```diff
@@ -716,10 +716,30 @@ public static String applyUrlScheme(final String url, final String urlScheme) {
     return (at == -1) ? (urlScheme + "://" + url) : urlScheme + url.substring(at);
   }
 
+  /**
```

Comment:
[+1] Awesome Javadocs, thanks for adding these.

[0] Definitely not a job for this PR, but I think we could make SolrJ easier to understand if we were a bit tighter in our URL/path terminology.    "Base" means a lot of different things to a lot of different people.

You could imagine defining terms like "root URL" = `scheme://host:port/$contextRoot`, "API base URL" = `$root/api` (v2), `$root/solr` (v1), "collection URL" = ...

Just thinking aloud.  This is all very minor, but I've spent a lot of time lately looking at how SolrJ builds these APIs so it's on my mind.